### DISCUSSION TITLE
Storing meta data for measurement in asammdf

### DISCUSSION THREAD

**Discussion by mimxrt** [danielhrisca/asammdf#508] [Q&A] [answered]
Hello everyone,

From the [ASAM MDF standard wiki](https://www.asam.net/standards/detail/mdf/wiki/#c481) I read that an MD block should be used to store metadata in XML format. However, I can't find any way to store such information (e.g. configuration parameters during the measurement valid for all data groups or channel groups) with asammdf. Is it because the [asammdf documentation](https://asammdf.readthedocs.io/en/latest/intro.html?highlight=xml#major-features-not-implemented-yet) states

> xml schema for MDBLOCK: most metadata stored in the comment blocks will not be available

or am I looking in the wrong place? How can I store additional key-value metadata in addition to the typical time series (`Signal`) data?

Thank you!

**Comment by danielhrisca** [comment, marked as answer]
MDBLOCKs can be used as comments for each other block type but it will contains different default xml tags depending on the block type. You can of course just insert your custom tags in the xml string

for example for the file header comment <HDcomment> and <TX> are mandatory, <MYTAG> is custom:

```xml
<HDcomment>
<TX>mandatory tag</TX>
<MYTAG> this is just for me</MYTAG>
</HDcomment>
```

**Reply by mimxrt** [reply]
Thank you for your reply. I understand that I can set, e.g., `mdf.header.comment = "something"` and it will be stored as
```
<HDcomment>
<TX>something</TX>
</HDcomment>
```
However, how would I add `<MYTAG> this is just for me</MYTAG>` to the MDF file from your example?

**Reply by danielhrisca** [reply]
this is the way to go:

```python
mdf.header.comment = """<HDcomment>
<TX>mandatory tag</TX>
<MYTAG> this is just for me</MYTAG>
</HDcomment>"""
```

So you need to construct the XML string yourself. There is no fancy method to create the XML and set set the tags

**Reply by mimxrt** [reply]
Understood, thank you very much!
